### DISCUSSION TITLE
Tps drops after 50 mspt

### DISCUSSION THREAD

**Discussion by MCPLAYER127272** [PurpurMC/Purpur#1426] [__Dismissed__] [answered]
How to make sure that the server does not go to bed after 50MSPT
The processor and RAM are not used to the maximum, while when the MSPT reaches 50, the TPS goes down. 
I heard that this is how it should be. But I saw how MSPT grew to 200 on other projects and everything was fine. 
The core of purpur
CPU 4x 4.9ghz
Ram 10 GB

**Comment by hexadecimal233** [comment]
intended behavior.
you can check wiki for minecraft tick mechanism

**Comment by BillyGalbreath** [comment, marked as answer]
I don't think you are fully understanding what MSPT and TPS are. Let me clarify:

MSPT means **M**illi**S**econds **P**er **T**ick. It's the amount of time that a tick takes to fully process.

TPS means **T**icks **P**er **S**econd. It's how many ticks were processed in a full second (1000 milliseconds).

If you take a full second (1000 milliseconds) and divide it by 50 (the maximum amount of time a tick should take), you get 20. So at most, you should have 20 ticks processed in a full second.

If a tick takes _less_ than 50 milliseconds to process, then the game will wait until 50 milliseconds have passed before starting to process the next tick. This ensures the game runs at a steady 20 ticks per second as described above.

If a tick takes _more_ than 50 milliseconds, then you will have _less_ ticks fit into 1000 milliseconds. Let's say the ticks take 200 milliseconds, as you described. 1000 divided by 200 is 5. Meaning you will have a max TPS of 5.

I hope this helps clarify what MSPT and TPS are.

**Reply by MCPLAYER127272** [reply]
Thank you, but how to make sure that the processor is used at 100

**Comment by CatTeaA** [comment]
MSPT×TPS≤1000

MSPT≤50，TPS=20

MSPT＞50，TPS=1000÷MSPT

**Comment by Potothingi** [comment]
> I heard that this is how it should be. But I saw how MSPT grew to 200 on **other projects** and everything was fine.

Can you tell me about your other projects?
It's amazing. Made the impossible possible. :p

**Reply by MCPLAYER127272** [reply]
Mehaniks- it's Russia server...
